File: Skill II/Day5/Part2/SolutionFromLeetcode.py
class Solution(object):
    def addToArrayForm(self, A, K):
        A[-1] += K
        for i in xrange(len(A) - 1, -1, -1):
            carry, A[i] = divmod(A[i], 10)
            if i: A[i-1] += carry
        if carry:
            A = map(int, str(carry)) + A
        return A

# Runtime: 350 ms, faster than 46.76% of Python online submissions for Add to Array-Form of Integer.
# Memory Usage: 13.9 MB, less than 57.18% of Python online submissions for Add to Array-Form of Integer.

File: Skill II/Day5/Part2/SolutionFromLeetCode.java
class Solution {
    public List<Integer> addToArrayForm(int[] A, int K) {
        int N = A.length;
        int cur = K;
        List<Integer> ans = new ArrayList();

        int i = N;
        while (--i >= 0 || cur > 0) {
            if (i >= 0)
                cur += A[i];
            ans.add(cur % 10);
            cur /= 10;
        }

        Collections.reverse(ans);
        return ans;
    }
}

// Runtime: 4 ms, faster than 89.25% of Java online submissions for Add to Array-Form of Integer.
// Memory Usage: 43.6 MB, less than 96.33% of Java online submissions for Add to Array-Form of Integer.